manifests/plugin/dynflow.pp

```diff
@@ -54,12 +54,14 @@
     ~> file { '/etc/smart_proxy_dynflow_core/settings.yml':
       ensure  => file,
       content => template('foreman_proxy/plugin/dynflow_core.yml.erb'),
+      notify  => Service['smart_proxy_dynflow_core'],
     }
-    ~> file { '/etc/smart_proxy_dynflow_core/settings.d':
+    -> file { '/etc/smart_proxy_dynflow_core/settings.d':
```

Comment:
I think `->` doesn't make a whole lot of sense after these changes. Perhaps it should be
```puppet
    foreman_proxy::plugin { 'dynflow_core':
      notify  => Service['smart_proxy_dynflow_core'],
    } 

    file { '/etc/smart_proxy_dynflow_core/settings.yml':
      ensure  => file,
      content => template('foreman_proxy/plugin/dynflow_core.yml.erb'),
      require => Foreman_proxy::Plugin['dynflow_core'],
      notify  => Service['smart_proxy_dynflow_core'],
    } 

    file { '/etc/smart_proxy_dynflow_core/settings.d':
      ensure  => link,
      target  => "${foreman_proxy::config_dir}/settings.d",
      require => Foreman_proxy::Plugin['dynflow_core'],
      notify  => Service['smart_proxy_dynflow_core'],
    } 

    systemd::service_limits { 'smart_proxy_dynflow_core.service':
      limits          => { 
        'LimitNOFILE' => $open_file_limit,
      },
      restart_service => false,
      require         => Foreman_proxy::Plugin['dynflow_core'],
    } 

    service { 'smart_proxy_dynflow_core':
      ensure => running,
      enable => true,
    } 
```

I also think this is why usually people use the `install`, `config` and `service` pattern.

Reply:
Yeah, most of those -> can go, you're right

Reply:
Would you mind updating it? Then I'm fine with merging it (and cherry picking).